src/shapes/text.class.js

```diff
@@ -373,6 +373,7 @@
         this.setPathInfo();
       }
       this.__skipDimension = false;
+      this.styles = fabric.util.stylesFromArray(this.styles, this.text);
```

Comment:
As mention earlier I don't think it's safe to set styles before they are parsed (I would remove L368) and combine it here as follows.
I doesn't seem that logic uses it up to here
```suggestion
      this.styles = fabric.util.stylesFromArray(options ? (options.styles || { }) : { }, this.text);
```

Reply:
good call